Smash_Combos.Core/Cqrs/Infractions/PutInfraction/PutInfractionRequest.cs

```diff
@@ -0,0 +1,18 @@
+﻿using MediatR;
+using Smash_Combos.Domain.Models;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Smash_Combos.Core.Cqrs.Infractions.PutInfraction
+{
+    public class PutInfractionRequest : IRequest<PutInfractionResponse>
+    {
+        public int Id { get; set; }
+        public int? BanDuration { get; set; }
+        public int? Points { get; set; }
+        public InfractionCategory Category { get; set; }
+        public string Body { get; set; }
+        public bool LiftBan { get; set; }
```

Comment:
How does LiftBan work? How do we pass this request on the frontend? Correct me if I’m wrong but doesn’t LiftBan need to be in the Domain on the `Infraction` model to be able to pass it as `true` through the controller?

Should LiftBan be set to `private set` to prevent any users from doing some sneaky API requests to unban themselves?

Reply:
We don't need to have LiftBan in the `Infraction-Model`, we only use the property on the `PutInfractionRequest`-Object that we expect in the `PutInfraction`-Method in the Controller.
You could think of it this way:
We no longer send the _object to change_ with the http-Request, we just send _the data we want changed_. The `PutInfractionRequest` contains the data that we want changed (Category, Body, Points, etc.) and a property to show if we want to lift the ban. 

In the `PutInfractionRequestHandler`-Class we can see how `LiftBan` works:
```
infraction.BanDuration = request.BanDuration;
infraction.Points = DeterminePoints(request);
infraction.Category = request.Category;
infraction.Body = request.Body;

if (request.LiftBan)
      infraction.BanLiftDate = DateTime.Now;
```
Basically the liftDate just gets set to the current time if the `LiftBan`-Property is true! There's no need to add it to the `Infraction`-class.

As for security, making it `private set` won't help too much (I'll still do it for good measure), as I think the biggest risk would be users _intercepting_ the http request before it gets to the server and changing the request-body, which wouldn't be stopped by the `private set`. I propose the following:
When a PutInfractionRequest comes in, we get the _current user ID_ and check if the currently logged in user is a Moderator or an Admin, and if he isn't, we deny the request. That way, even if the request is tampered with, as long as the user is not a mod or admin, nothing will happen.